sound/soc/sof/intel/hda.c

```diff
@@ -976,6 +976,10 @@ void hda_dsp_remove(struct snd_sof_dev *sdev)
 	if (sdev->dspless_mode_selected)
 		goto skip_disable_dsp;
 
+	/* Cancel the microphone privacy work if it is active */
+	if (hda->mic_privacy.active)
+		cancel_work_sync(&hda->mic_privacy.work);
```

Comment:
as long as the container structure is 0-initialised (e.g. allocated per `kcalloc()` or similar) this won't represent a bisection breakage, because the `if` above won't be entered, but still it's a strange split of commits - cancel work here that's initialised and submitted in the next commit... I'm just merge the two.

Reply:
I think a single patch is preferable but not a deal-breaker for me